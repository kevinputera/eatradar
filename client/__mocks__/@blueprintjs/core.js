/**
 * This manual mock only mocks Toaster, Position, and Intent component
 * and is only used in StatusMessage.test.js
 */
const core = jest.requireActual('@blueprintjs/core');

const clearMock = jest.fn();
const showMock = jest.fn();
const Toaster = {
  create: () => ({
    clear: clearMock,
    show: showMock,
  }),
};

module.exports = {
  ...core,
  clearMock,
  showMock,
  Toaster,
};
