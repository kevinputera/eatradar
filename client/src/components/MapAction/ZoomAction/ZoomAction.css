.zoom-action {
  display: flex;
  flex-direction: column;
}
