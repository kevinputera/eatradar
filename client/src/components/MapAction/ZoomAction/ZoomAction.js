import React from 'react';
import { Button } from '@blueprintjs/core';

import './ZoomAction.css';

function ZoomAction(props) {
  return (
    <div className="zoom-action">
      <div className="zoom-control">
        <Button
          large
          icon="small-plus"
          onClick={props.handleZoomIn}
          style={{ borderRadius: '20px' }}
        />
      </div>
      <div className="zoom-control">
        <Button
          large
          icon="small-minus"
          onClick={props.handleZoomOut}
          style={{ borderRadius: '20px' }}
        />
      </div>
    </div>
  );
}

export default ZoomAction;
