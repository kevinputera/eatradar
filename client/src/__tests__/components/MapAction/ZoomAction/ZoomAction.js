import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Button } from '@blueprintjs/core';

import ZoomAction from '../../../../components/MapAction/ZoomAction/ZoomAction';

describe('tests for ZoomAction component', () => {
  test('render must match snapshot', () => {
    const tree = create(<ZoomAction />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('zoom-in click must trigger handleZoomIn prop', () => {
    const handleZoomIn = jest.fn();
    const wrapper = shallow(<ZoomAction handleZoomIn={handleZoomIn} />);
    wrapper.find('.zoom-in-button').simulate('click');
    expect(handleZoomIn).toHaveBeenCalledTimes(1);
  });

  test('zoom-out click must trigger handleZoomOut prop', () => {
    const handleZoomOut = jest.fn();
    const wrapper = shallow(<ZoomAction handleZoomOut={handleZoomOut} />);
    wrapper.find('.zoom-out-button').simulate('click');
    expect(handleZoomOut).toHaveBeenCalledTimes(1);
  });
});
