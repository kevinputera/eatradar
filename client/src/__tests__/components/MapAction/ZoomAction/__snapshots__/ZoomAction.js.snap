// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for ZoomAction component render must match snapshot 1`] = `
<div
  className="zoom-action"
>
  <div
    className="zoom-control"
  >
    <button
      className="bp3-button bp3-large zoom-in-button"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      style={
        Object {
          "borderRadius": "20px",
        }
      }
      type="button"
    >
      <span
        className="bp3-icon bp3-icon-plus"
        icon="plus"
      >
        <svg
          data-icon="plus"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <desc>
            plus
          </desc>
          <path
            d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
  <div
    className="zoom-control"
  >
    <button
      className="bp3-button bp3-large zoom-out-button"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      style={
        Object {
          "borderRadius": "20px",
        }
      }
      type="button"
    >
      <span
        className="bp3-icon bp3-icon-minus"
        icon="minus"
      >
        <svg
          data-icon="minus"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <desc>
            minus
          </desc>
          <path
            d="M13 7H3c-.55 0-1 .45-1 1s.45 1 1 1h10c.55 0 1-.45 1-1s-.45-1-1-1z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
